File: parsers/ruby-rubyzip/unzip.rb
require 'zip'

Zip.on_exists_proc = true
Dir.chdir(ARGV[1])

Zip::File.open(ARGV[0]) do |zip_file|
  zip_file.each do |entry|
    entry_path = File.join(Dir.pwd, entry.name)
    FileUtils.mkdir_p(File.dirname(entry_path))
    entry.extract
  end
end

File: parsers/34-nodejs-jszip/unzip.js
const { loadAsync } = require('jszip');
const { dirname } = require('path');
const { readFile, mkdir, writeFile } = require('fs/promises');

(async () => {
  const file = await readFile(process.argv[2]);
  const zip = await loadAsync(file);

  process.chdir(process.argv[3]);

  for (const entry of Object.values(zip.files)) {
    if (entry.dir) {
      await mkdir(entry.name, { recursive: true });
    } else {
      await mkdir(dirname(entry.name), { recursive: true });
      const content = await entry.async('nodebuffer');
      await writeFile(entry.name, content);
    }
  }
})();
